### PR TITLE
Magikarp jump temp battles

### PR DESCRIPTION
**Pull request by Jaaslet**
- Added magikarp jump temp battles
- Gave them the correct fishies
- The give the fishies, if no gym has it and no temp battle before it
- Added the correct requirements for them
- Added them to towns

Still missing balance and defeat text

### CODE REVIEW

File: src/scripts/Update.ts [hunk 1]
```diff
@@ -1559,6 +1559,29 @@ class Update implements Saveable {
                 }
             }
         },
+        '0.10.5': ({ playerData, saveData }) => {
+            saveData.statistics.temporaryBattleDefeated = Update.moveIndex(saveData.statistics.temporaryBattleDefeated, 159);
```
Comment (Aegyo) [inline src/scripts/Update.ts:1563]: We do these massive moveIndex walls so often, maybe we want a `Update.moveIndexRange(159, 179)`

Comment (RedSparr0w) [reply src/scripts/Update.ts:1563]: Could only really be used for new indexes, so probably something like `insertNewIndexRange`

**Review by RedSparr0w — APPROVED** [review]
Looks good to me 👍 

Just a couple questions, don't we have badges implemented for the sub region?
Shouldn't these be gyms?
Or are the badges coming from somewhere else?

**Comment by Jaaslet** [comment]
> Just a couple questions, don't we have badges implemented for the sub region? Shouldn't these be gyms? Or are the badges coming from somewhere else?

I have added 10 gyms in an other pr. Made these as temp battles because 30+ gyms felt like too many

**Pull request merged**